core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java

```diff
@@ -724,16 +756,24 @@ void resetStateIfNecessary() {
             // Second, check if writer's structure version has changed
             // compared with the known metadata.
             if (latestKnownMetadata != null) {
-                long metadataVersion = writerAPI.getMetadataVersion();
-                if (latestKnownMetadata.getMetadataVersion() != metadataVersion) {
+                if (latestKnownMetadataVersion != writerAPI.getMetadataVersion()) {
                     // clear() frees latestKnownMetadata and sets it to null
                     clear();
+                    latestKnownMetadata = null;
                 }
             }
             if (latestKnownMetadata == null) {
                 // Get the latest metadata.
                 try {
-                    latestKnownMetadata = engine.getLegacyMetadata(tableToken);
+                    if (writerIsLocal) {
```

Comment:
is this not the same as `isWal()`?
if yes, `writerIsLocal` could be removed.

Reply:
Yes, it's same. I removed `writerIsLocal`